src/site/content/en/blog/enabling-shared-array-buffer/index.md

```diff
@@ -0,0 +1,92 @@
+
+A general mitigation is to ensure a webpage's system process doesn't contain sensitive data from elsewhere. Chrome had invested in a mutliprocess architecture from the start ([remember the comic?](https://www.google.com/googlebooks/chrome/big_00.html)), but there were still cases where data from multiple sites could end up in the same process:
+
+```html
+<iframe src="https://your-bank.example/balance.json"></iframe>
+<script src="https://your-bank.example/balance.json"></script>
+<link rel="stylesheet" href="https://your-bank.example/balance.json" />
+<img src="https://your-bank.example/balance.json" />
+<video src="https://your-bank.example/balance.json"></video>
+<!-- …and more… -->
+```
+
+These APIs have a 'legacy' behavior that allows content from other origins to be used without opt-in from the other origin. These requests are made with the cookies of the other origin, so it's a full 'logged in' request. Nowadays, new APIs require the other origin to opt-in using [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+
+We worked around these legacy APIs by preventing content from entering the webpage's process if it looked 'incorrect', and called it [cross-origin read blocking](https://developers.google.com/web/updates/2018/07/site-isolation#corb). So, in the above cases, we wouldn't allow JSON to enter the process, as it isn't a valid format for any of those APIs. That is, except iframes. For iframes we put the content in a different process.
+
+With these mitigations in place, we reintroduced `SharedArrayBuffer` in Chrome 68 (July 2018), but only on desktop. The extra process requirements meant we couldn't do the same on mobile devices. It was also noted that Chrome's solution was incomplete, as we were only blocking 'incorrect' data formats, whereas it's possible (although unusual) that valid CSS/JS/images at guessable URLs can contain private data.
+
+Web standards folks got together to come up with a more complete solution. The solution was to give pages a way to say "I hereby relinquish my ability to bring other-origin content into this process without their opt-in". This declaration is done via [COOP and COEP headers](/coop-coep/) served with the page. The browser enforces that, and in exchange the page gains access to `SharedArrayBuffer` and other APIs with similar powers. Other origins can opt-in to content embedding via [`Cross-Origin-Resource-Policy`](<https://developer.mozilla.org/en-US/docs/Web/HTTP/Cross-Origin_Resource_Policy_(CORP)>) or [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS).
+
+Then, in December 2020, I wrote this article, and you read it. Hello.
+
+And that's where we are now. Chrome 88 brings `SharedArrayBuffer` back to Android for pages that are cross-origin isolated, and Chrome 91 brings the same requirements to desktop, both for consistency, and to achieve total cross-origin isolation.
```

Comment:
To be clear, only Chrome on Android and not WebView or CCT

Reply:
Ohh, so `SharedArrayBuffer` won't work in CCT? Does it work in PWAs added to homescreen (via the APK method)?

Reply:
Sorry for the confusion! 
Chrome on Android and therefore CCT as well will support COOP/COEP and doing so SABs are available.
WebView supports only COOP yet. COEP discussion is ongoing,but it's unlikely to be shipped soon

Reply:
TL;DR: WebView's process model sucks.